### PR TITLE
updateFormData

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
This is the first step in addressing https://github.com/department-of-veterans-affairs/vets.gov-team/issues/18161

## Testing done
Wrote me some unit tests. :man_farmer: 

## Screenshots
N/A

## Acceptance criteria
- [x] The existing functionality is not affected
- [x] New `updateFormData` callback is handled both on a normal form page as well as the review page

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx [hunk 1]
```diff
@@ -170,7 +170,13 @@ export default class ReviewCollapsibleChapter extends React.Component {
                       )
                     }
                     onChange={formData =>
-                      this.onChange(formData, page.arrayPath, page.index)
+                      this.onChange(
+                        typeof page.updateFormData === 'function'
```
Comment (jbalboni) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx]: Think there's a way to get this logic into a function that can be shared? It's in three different places now in a similar-ish form. But they might not quite be close enough.

Comment (cvalarida) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx]: Yeah, like we talked about earlier, we would ideally be able to consolidate this in the `SET_DATA` reducer, but I think there's a lot more overhead with that approach. (Getting the `updateFormData` function to only fire when data on its page changes—or just always fire all `updateFormData` functions for all pages on all data changes :grimacing:—getting the old data separate from the new data, that kind of thing.) I didn't really look too much into it, though.

I'm hoping with sufficient testing, it'll be alright. :crossed_fingers:

**Review by jbalboni — APPROVED** [review]
LGTM, can you also create a ticket to add this function to the forms documentation in the vfs tools repo?

**Comment by cvalarida** [comment]
[Done!](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/18735) Thanks for the review. :raised_hands:

**Pull request merged**